provider/internal/java/dependency.go

```diff
@@ -134,9 +134,7 @@ func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, locatio
 	p.log.V(10).Info("Analyzing POM",
 		"POM", fmt.Sprintf("%s:%s:%s", pomCoordinate(pom.GroupID), pomCoordinate(pom.ArtifactID), pomCoordinate(pom.Version)),
 		"error", err)
-	if err != nil {
```

Comment:
Note to reviewer: this is because this err is checked above this is a redundant call